### PR TITLE
added solution hb-05

### PR DESCRIPTION
**Pull request by IgorBochkor**
added solution hb-05

### CODE REVIEW

**Review by yurwing — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/User.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public void setSalt(byte[] salt) {
+        this.salt = salt;
+    }
+
+    @Override
+    public String toString() {
+        return "User{"
+                + "id=" + id
+                + ", email='" + email + '\''
+                + ", password='" + password + '\''
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/model/User.java]: let's not make secure data visible

File: src/main/java/com/dev/cinema/security/AuthenticationService.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.dev.cinema.security;
+
+import com.dev.cinema.exception.AuthenticationException;
+import com.dev.cinema.model.User;
+
+public interface AuthenticationService {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/security/AuthenticationService.java]: redundant empty line

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userFindByEmail = userService.findByEmail(email);
+        if (userFindByEmail.isPresent()) {
+            User user = userFindByEmail.get();
+            String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+            if (hashPassword.equals(user.getPassword())) {
+                return user;
+            }
+        }
+        throw new AuthenticationException("Can't get login by email " + email);
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: let's make the message more informative
```suggestion
        throw new AuthenticationException("The login or password is incorrect");
```

File: src/main/java/com/dev/cinema/util/HashUtil.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+
+public class HashUtil {
+    private static final String SHA_512_ALGORITHM = "SHA-512";
+
+    private HashUtil() {
+
```
Comment (KseniiaMakarova) [inline src/main/java/com/dev/cinema/util/HashUtil.java]: redundant empty line

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/model/User.java (hunk 1 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/model/User.java]: hide password

File: src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java (hunk 3 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/cinema/security/AuthenticationServiceImpl.java]: fixed

**Review by KseniiaMakarova — APPROVED** [review]

**Pull request merged**